pkg/issuer/acme/dns/route53/route53.go

```diff
@@ -297,18 +292,15 @@ func newTXTRecordSet(fqdn, value string, ttl int) *route53.ResourceRecordSet {
 // avoid spurious challenge updates.
 //
 // The given error must not be nil. This function must be called everywhere
-// we have a non-nil error coming from an aws-sdk-go func.
+// we have a non-nil error coming from an aws-sdk-go func. The passed error
+// is modified in place. This function does not work in case the full error
+// message is pre-generated at construction time (instead of when Error() is
+// called), which is the case for eg. fmt.Errorf("error message: %w", err).
```

Comment:
As far as I know it creates a new error with an `Unwrap` method that allows the underlying error to be accessed.
Perhaps I misunderstood how `%w` works. 
 * https://go.dev/blog/go1.13-errors#wrapping-errors-with-w

Reply:
This is the message string that is created when you call the function: https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/fmt/errors.go;l=55
If you change the wrapped error afterwards, the message will still be the same.

Reply:
Ah, I think I understand. I hadn't looked at the implementation of `fmt.Errorf` until now.
And I see that it sets the error message string when it is called, so any subsequent changes to the wrapped error are not reflected in the error string....like you said in that comment:

https://github.com/golang/go/blob/1400b268529f1bbacde83b3dc022fec0b196846b/src/fmt/errors.go#L26-L32